components/src/dynamo/common/utils/video_utils.py

```
@@ -242,3 +245,297 @@ def encode_to_video_bytes(
+) -> bytes:
+    """Encode via imageio -> ffmpeg (NVENC path). Preserves existing behavior."""
+    try:
+        import imageio.v3 as iio
+    except ImportError:
+        try:
+            import imageio as iio  # type: ignore[no-redef]
+        except ImportError:
+            raise ImportError(
+                "imageio is required for video encoding. "
+                "Install with: pip install imageio[ffmpeg]"
+            )
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)